### PR TITLE
debuginfoproviders and settings available for frontends

### PR DESCRIPTION
**Pull request by an-dr-eas-k**
Let me try again with reworked version of what I showed you yesterday.

Again this pull request solves the issues in #1676 and #1677. This time with less restructuring and considering your commit be887d0.

Regards,

### CODE REVIEW

**Comment by siegfriedpammer** [comment]
If you don't mind, I will refactor your proposed changes a bit. (That will include removing some of your commits that remove code, which you later revert...)

But, I am not sure when I will have time for this. So please be patient.

**Comment by christophwille** [comment]
New PR by @siegfriedpammer has superseded this one, closing.
